packages/consumption/src/modules/requests/itemProcessors/readAttribute/ReadAttributeRequestItemProcessor.ts

```diff
@@ -55,12 +178,19 @@ export class ReadAttributeRequestItemProcessor extends GenericRequestItemProcess
         requestInfo: LocalRequestInfo
     ): Promise<ReadAttributeAcceptResponseItem> {
         const parsedParams = AcceptReadAttributeRequestItemParameters.from(params);
+        let sharedLocalAttribute;
 
-        let sharedLocalAttribute: LocalAttribute;
         if (parsedParams.isWithExistingAttribute()) {
             sharedLocalAttribute = await this.copyExistingAttribute(parsedParams.existingAttributeId, requestInfo);
-        } else {
-            sharedLocalAttribute = await this.createNewAttribute(parsedParams.newAttribute!, requestInfo);
+        } else if (parsedParams.isWithNewAttribute()) {
+            if (parsedParams.newAttribute.owner.equals("")) {
+                parsedParams.newAttribute.owner = this.currentIdentityAddress;
+            }
+            sharedLocalAttribute = await this.createNewAttribute(parsedParams.newAttribute, requestInfo);
+        }
+
+        if (!sharedLocalAttribute) {
+            throw new Error("this should never happen");
```

Comment:
Maybe add a check in the beginning whether isWithExistingAttribute or isWithNewAttribute is true, so this could be omitted.